frame/treasury/src/lib.rs

```diff
@@ -482,18 +479,17 @@ decl_module! {
 		/// # <weight>
 		/// - Complexity: `O(R + T)` where `R` length of `reason`, `T` is the number of tippers.
 		///   - `O(T)`: decoding `Tipper` vec of length `T`
-		///     `T` is less than `MAX_TIPPERS_COUNT`, it is charged as maximum and then refund.
+		///     `T` is charged as upper bound given by `ContainsCountUpperBound`.
 		///     The actual cost depends on the implementation of `T::Tippers`.
 		///   - `O(R)`: hashing and encoding of reason of length `R`
 		/// - DbReads: `Tippers`, `Reasons`
 		/// - DbWrites: `Reasons`, `Tips`
 		/// # </weight>
-		#[weight = 107_700_000 + 4_000 * reason.len() as u64 + 481_000 * MAX_TIPPERS_COUNT
+		#[weight = 110_000_000
+			+ 4_000 * reason.len() as u64
```

Comment:
I notice I am surprised by the number `4_000` here, it seems very small. Just verifying it's not a type or something :-)

Reply:
Yes better to double check. but it is correct.

reason is hashed, decoded and encoded, but it is a `Vec<u8>` so encoding it is actually optimized in parity-scale-codec.
And it is correct with the 4_000 coef in report_awesome.